### DISCUSSION TITLE
Confusion about webhook config creation

### DISCUSSION THREAD

**Discussion by allenhaozi** [crossplane/crossplane#3583] [Q&A]
### What problem are you facing?

Hey crossplane developer:

I am reading and learning the code of crossplane, I like the elegant style very much

But in the process of reading, there is one place that is not very clear,

Why create a webhook config here by loading [local files](https://github.com/crossplane/crossplane/blob/master/internal/initializer/webhook_configurations.go#L88) instead of helm charts 
for example [secret](https://github.com/crossplane/crossplane/blob/master/cluster/charts/crossplane/templates/secret.yaml)

Because I found that if it is uninstalled, the webhook config is still there,

did I overlook something? thanks

**Comment by hasheddan** [comment]
Hey @allenhaozi! Crossplane uses an init container for CRDs and webhooks to work around some of the limitations in `helm upgrade`. You can see the motivation in the original issue here: https://github.com/crossplane/crossplane/issues/2156. I am going to convert this to a discussion for any further conversation as I don't think there are any immediate action items required :)

**Comment by allenhaozi** [comment]
thank you @hasheddan for your reply
got it
